### PR TITLE
initial structure redis

### PR DESCRIPTION
**Pull request by aathiraashok**

### CODE REVIEW

File: spec.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+				Default:     "",
+				Required:    true,
+				Description: "port to the redis source",
+			},
+			config.ConfigKeyKey: {
+				Default:     "",
+				Required:    true,
+				Description: "key name for connector to read.",
+			},
+			config.ConfigKeyDatabase: {
```
Comment (amansardana) [inline spec.go:68]: Is database name a required field, Can we connect to a key without it?

File: source/iterator.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package source
+
+import (
+	"context"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	redis "github.com/gomodule/redigo/redis"
+)
+
+func NewCDCIterator(ctx context.Context, client redis.Conn, key string) (*sdk.Record, error) {
```
Comment (amansardana) [inline source/iterator.go]: This should return an instance of a struct which adheres to the iterator interface and not sdk.Record

File: config/config_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+				ConfigKeyPassword: "12345678",
+				ConfigKeyChannel:  "my_channel",
+			},
+			want: Config{},
+			err:  fmt.Errorf("key config value must be set"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got, _ := Parse(tt.config); !reflect.DeepEqual(got, tt.want) {
```
Comment (amansardana) [inline config/config_test.go]: Where are we checking the `error`?

File: source/iterator.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+					return
+				}
+			case error:
+				return
+			}
+		}
+	}()
+	go func() {
+		defer wg.Done()
+		psc.Subscribe(key)
```
Comment (amansardana) [inline source/iterator.go]: Doesn't it return any error?

File: source/source.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	sdk.UnimplementedSource
+
+	config   config.Config
+	client   redis.Conn
+	iterator Iterator
+}
+type Iterator interface {
+	HasNext(ctx context.Context) bool
+	Next(ctx context.Context) (sdk.Record, error)
+	// Stop()
```
Comment (amansardana) [inline source/source.go]: Why is it commented out?

Comment (amansardana) [reply source/source.go]: In Stop, close the go routine that we started, in NewCDCIterator

File: source/iterator.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	redis "github.com/gomodule/redigo/redis"
+)
+
+type CDCIterator struct {
+	client     *redis.Conn
+	key        string
+	psc        redis.PubSubConn
+	testRecord map[string]sdk.Record
```
Comment (amansardana) [inline source/iterator.go]: change the name

File: source/iterator.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+	wg.Wait()
+	return cdc, nil
+}
+func (i *CDCIterator) HasNext(ctx context.Context) bool {
+	return len(i.testRecord) > 0
+}
+
+func (i *CDCIterator) Next(ctx context.Context) (sdk.Record, error) {
+	val := i.testRecord[i.key]
+	delete(i.testRecord, i.key)
```
Comment (amansardana) [inline source/iterator.go]: Will we have only one message in map or multiple?

File: source/iterator.go [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+		quit:    make(chan bool),
+	}
+	go func() {
+		select {
+		case <-cdc.quit:
+			return
+		default:
+
+		}
+		for {
```
Comment (viknesh-nm) [inline source/iterator.go]: for should be in default.

File: source/iterator.go [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+			switch n := psc.Receive().(type) {
+			case redis.Message:
+				data := sdk.Record{
+					Payload: sdk.RawData(n.Data),
+				}
+				sdk.Logger(ctx).Info().
+					Str("channel", n.Channel).
+					Str("data", string(n.Data)).
+					Msg("message")
+				// c := map[string]sdk.Record{
```
Comment (viknesh-nm) [inline source/iterator.go]: remove, if its not necessary

File: source/iterator.go [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+			case redis.Subscription:
+				if n.Count == 0 {
+					return
+				}
+			case error:
+				return
+			}
+		}
+	}()
+	errs := make(chan error)
```
Comment (viknesh-nm) [inline source/iterator.go]: make the channel as buffered to `make(chan error, 1)`

**Review by viknesh-nm — CHANGES_REQUESTED** [review]

**Pull request closed without merging**